reactor-core/src/main/java/reactor/core/publisher/FluxScanSeed.java

```diff
@@ -61,21 +59,96 @@
 
 	@Override
 	public void subscribe(Subscriber<? super R> s) {
-		R initialValue;
+		ScanSeedCoordinator<T, R> coordinator =
+				new ScanSeedCoordinator<>(s, source, accumulator, initialSupplier);
+
+		s.onSubscribe(coordinator);
 
-		try {
-			initialValue = Objects.requireNonNull(initialSupplier.get(),
-					"The initial value supplied is null");
+		if (!coordinator.isCancelled()) {
+			coordinator.onComplete();
 		}
-		catch (Throwable e) {
-			Operators.error(s, Operators.onOperatorError(e));
-			return;
+	}
+
+	static final class ScanSeedCoordinator<T, R>
+			extends Operators.MultiSubscriptionSubscriber<R, R> {
+
+		final         Supplier<R>                 initialSupplier;
+		private final Publisher<? extends T>      source;
```

Comment:
the 'source' should be package-private instead of private?

(otherwise for alignment I just let IntelliJ auto-format after importing the code style)